lib/js/jsconsole.nim

```diff
@@ -67,4 +69,37 @@ proc timeLog*(console: Console, label = "".cstring) {.importcpp.}
 proc table*(console: Console) {.importcpp, varargs.}
   ## https://developer.mozilla.org/docs/Web/API/Console/table
 
+since (1, 5):
+  # proc nor func can not be used here, because may eval assertion at compile-time,
+  # writing directly false or true to the JavaScript instead of the expression.
+  # importjs nor importcpp can not be used with template.
+  type InstantiationInfo = tuple[filename: string, line: int, column: int]
+
+  proc `$`(info: InstantiationInfo): string =
+    var temp = ""
+    temp.toLocation(info.filename, info.line, info.column + 1)
+    result.addQuoted(temp)
+
+  template jsAssert*(console: Console; assertion) =
+    ## JavaScript `console.assert`, for NodeJS this prints to stderr,
+    ## assert failure just prints to console and do not quit the program,
+    ## this is not meant to be better or even equal than normal assertions,
+    ## is just for when you need faster performance *and* assertions,
+    ## otherwise use the normal assertions for better user experience.
+    ## https://developer.mozilla.org/en-US/docs/Web/API/Console/assert
+    runnableExamples: 
+      console.jsAssert(42 == 42) # OK
+      console.jsAssert(42 != 42) # Fail, prints "Assertion failed" and continues
+      console.jsAssert('`' == '\n' and '\t' == '\0') # Message correctly formatted
+      assert 42 == 42  # Normal assertions keep working
+
+    const
+      loc = instantiationInfo(fullPaths = compileOption("excessiveStackTrace"))
+      ploc = cstring($loc)
+    var msg = ""
+    msg.addQuoted(astToStr(assertion))
+    {.line: loc.}:
+      {.emit: ["console.assert(", assertion, ", ", ploc, ", ", msg.cstring, ");"].}
```

Comment:
the ploc + msg can be computed at CT, eg:
const msg = getMsg(astToStr(assertion), $loc).cstring
{.line: loc.}:
  {.emit: ["console.assert(", assertion, ", ", msg, ");"].}